### ISSUE TITLE
fix landXML parser so that it doesn't assume natural indices, uses acutal Point id values

### ISSUE THREAD

**Issue by Duke-of-Lizard** [emerald-geomodelling/EmeraldTriangles#18]
I have a terrain model file from a client whose point IDs do not start from 1, and there are missing point id somewhere in the file. for example: 
```
<Pnts><P id="2">7708128.575000000186 326970.457999999984 3.409</P><P id="3">7708104.796070420183 327008.606297355553 -0.2</P><P id="7">7708134.54899999965 326980.329000000027 4.552</P><P id="8">7708096.06045101583 327057.084694511432 0.8</P> 
```

However, our code assumes nice natural indices for points starting from 1. 

I've made changes to a new branch called 2023-11-landxml-use-point-ids to solve this: e641fc1426ebf57314449a3a8b951b34863db8e5

However, still two outstanding things I need help with: 
- [x] 1. By default, I want to preserve behaviour that the resultant data has a nice natural index. all this requires is a call to emeraldtriangles.cleanup.reindex: 
```
import emeraldtriangles.cleanup as cleanup
self.surface['vertices'], self.surface['triangles'] = cleanup.reindex(self.surface['vertices'], self.surface['triangles'])
```
However, I'm unsure how to properly pass a keyword argument to allow a user to turn this behaviour off and preserve the original vertex ids
- [ ] 2. the cython file `emeraldtriangles/io/_landxml.pyx` needs updating based on changes to `emeraldtriangles/io/landxml.py`, but I'm not familiar with cython syntax to do this on my own.

**Comment by Duke-of-Lizard** [comment]
Fixed the first item with this commit: 
60a90b674cd0b0bb0c3c321b3c7b873c1cb4af82

The Cython port can wait a little while, but not too long.

**Comment by Duke-of-Lizard** [comment]
I see that @redhog merged this pull request to fix this issue: 
https://github.com/emerald-geomodelling/EmeraldTriangles/pull/19

However, sadly, I don't think the port was successful. This is the output I would have expected from the private example file that I shared with you, @redhog :

![image](https://github.com/emerald-geomodelling/EmeraldTriangles/assets/61405744/5b06391a-a644-4c77-986b-bb3ccc2d51e9)
 But this is the result I get from the newly merged master version: 
 
![image](https://github.com/emerald-geomodelling/EmeraldTriangles/assets/61405744/5e54630f-db62-4b08-81e3-b69eaa80d4cd)

The points are in the right position, but something didn't quite go right with the indexing. Resultant vertices:
```
                   Y              X      Z   M
0       7.708129e+06  326970.458000  3.409 NaN
1       7.708105e+06  327008.606297 -0.200 NaN
2       7.708101e+06  327009.658287 -0.200 NaN
3       7.708130e+06  326972.241000  3.666 NaN
4       7.708096e+06  327056.063038  0.800 NaN
...              ...            ...    ...  ..
636887  7.767128e+06  354827.153100  5.026 NaN
636888  7.767139e+06  354827.540300  4.996 NaN
636889  7.767097e+06  354823.495900  4.116 NaN
636890  7.767096e+06  354826.517800  5.127 NaN
636891  7.767096e+06  354825.496600  5.104 NaN

[636892 rows x 4 columns]
```

resultant triangles:
```
              0       1       2
0             4       1       2
1             3       2       1
2             6       4       2
3            59      51       2
4            60      59       2
...         ...     ...     ...
1272986  636888  636887  636891
1272987  636891  636890  636888
1272988  636891  636889  636890
1272989  636893  636695  636694
1272990  636893  636894  636695

[1272991 rows x 3 columns]
```

The maximum valued vertex_index in the triangles array is greater than the maximum index in the vertices array. The vertices get set to natural-valued indices somewhere rather than preserving in the point indices within the input file

**Comment by Duke-of-Lizard** [comment]
FYI, when I disable the cython, here's the output I get: 

```
                   Y              X      Z   M      id
0       7.708129e+06  326970.458000  3.409 NaN       2
1       7.708105e+06  327008.606297 -0.200 NaN       3
2       7.708101e+06  327009.658287 -0.200 NaN       4
3       7.708130e+06  326972.241000  3.666 NaN       5
4       7.708096e+06  327056.063038  0.800 NaN       6
...              ...            ...    ...  ..     ...
636887  7.767128e+06  354827.153100  5.026 NaN  636891
636888  7.767139e+06  354827.540300  4.996 NaN  636892
636889  7.767097e+06  354823.495900  4.116 NaN  636893
636890  7.767096e+06  354826.517800  5.127 NaN  636894
636891  7.767096e+06  354825.496600  5.104 NaN  636895

[636892 rows x 5 columns]
```

triangles:
```
              0       1       2
0             3       0       1
1             2       1       0
2             5       3       1
3            58      50       1
4            59      58       1
...         ...     ...     ...
1272986  636885  636884  636888
1272987  636888  636887  636885
1272988  636888  636886  636887
1272989  636890  636692  636691
1272990  636890  636891  636692

[1272991 rows x 3 columns]
```

**Comment by Duke-of-Lizard** [comment]
Not sure what caused the weird cython issue before where the DataFrames had natural indices and triangles pointed to the old vertex ids. Seems to be working now, though. We caught a mismatch with the default keyword parameters in cython and python not being equal, but this commit resolved that: ac4823b300de893e7bfa616f3c470395b2db9a04

With that, this issue is closed
